src/main/java/org/dita/dost/writer/AbstractChunkTopicParser.java

```diff
@@ -300,7 +300,12 @@ Attributes processAttributes(final Attributes atts) {
                 // update @href to point back to current file
                 // if the location is moved to chunk, @href will
                 // be update again to the new location.
-                XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, relative + href);
+            	if(relative.toString().contains("#")) {
+            		//If the relative reference already has an anchor, we need to replace that with the current anchor.
+            		XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, relative.resolve(href).toString());
+            	} else {
+            		XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, relative + href);
+            	}
```

Comment:
This should probably use [`URLUtils.setFragment(URI, String)`](https://github.com/dita-ot/dita-ot/blob/develop/src/main/java/org/dita/dost/util/URLUtils.java#L551).

So something like

```java
final URI res = URLUtils.setFragment(relative, href.substring(1)); 
XMLUtils.addOrSetAttribute(resAtts, ATTRIBUTE_NAME_HREF, res);
```

`setFragment` should be improved so that it tests if the `fragment` argument starts with `#`. If so, it will strip it out.

Reply:
Thanks, I followed your advice and made the change. I did not make changes to URLUtils.setFragment yet, should I also do that and add an extra TC for the URLUtils.setFragment method?

Reply:
@raducoravu You don't have to make the change to automatically strip the fragment because this goes to bug fix release. We can do it in `develop` branch.